README.md
# BME680-Datalogger-with-EINK-display
Datalogger with BME680-Sensor, EINK Display, RTC and µSD-Storage

This project features a Arduino based datalogger reading temperature, relative humidity, pressure and IAQ values from a Bosch BME680 sensor. It displays the data at a 2.9 inch waveshare e-ink display using olikraus's u8g2 lib. The data is being synchronized with a DS3231 RTC and stored at a micro SD-Card.

The whole system runs on a adafruit feather M0 board

Please consider that this project is in a very early development status. I haven't completed the program code yet and I've just ordered the Mainboard.
If you want to make your own copy of this project, please contact me for information about recent unpublished changes.


Board/Changelog.txt

v2:
-fuel gauge (MAX17043
-delete tpl5110 option
add circuit to detect charging


v1:
-add battery voltage contact
-add button
-connected dc to GND(3-wire spi required -> change display jumper BS to 1)

-add tpl5110?
-Change to Feather M0
-add PFC8575C to expand boards (LEDs and buttons)



v0: Initial Release


Code/preliminary tests/v0/t_40.h
#define t_40_width 40
#define t_40_height 40
static unsigned char t_40_bits[] = {
   0x00, 0x00, 0x0e, 0x00, 0x00, 0x00, 0x80, 0x1f, 0x00, 0x00, 0x00, 0x80,
   0x3f, 0x00, 0x00, 0x00, 0xc0, 0x3b, 0x00, 0x00, 0x20, 0xc0, 0x31, 0x00,
   0x00, 0x20, 0xc0, 0x31, 0x00, 0x00, 0x20, 0xc0, 0x31, 0x00, 0x00, 0xf8,
   0xc1, 0x39, 0x00, 0x00, 0xf8, 0xc1, 0x39, 0x00, 0x00, 0x20, 0xc0, 0x31,
   0x00, 0x00, 0x20, 0xc0, 0x31, 0x00, 0x00, 0x20, 0xc0, 0x31, 0x00, 0x00,
   0x00, 0xc0, 0x31, 0x00, 0x00, 0x00, 0xc0, 0x33, 0x00, 0x00, 0xfe, 0xff,
   0xff, 0xff, 0x0f, 0x00, 0xc0, 0x3f, 0x00, 0x00, 0x00, 0xc0, 0x31, 0x00,
   0x00, 0x00, 0xc0, 0x31, 0x00, 0x00, 0x00, 0xc0, 0x31, 0x00, 0x00, 0x00,
   0xc0, 0x39, 0x00, 0x00, 0x00, 0xc0, 0x31, 0x00, 0x00, 0x00, 0xc0, 0x31,
   0x00, 0x00, 0x00, 0xc0, 0x31, 0xe0, 0x00, 0x00, 0xc0, 0x31, 0xf8, 0x07,
   0x00, 0xc0, 0x31, 0x0e, 0x0e, 0x00, 0xc0, 0x31, 0x03, 0x08, 0x00, 0xe0,
   0xf1, 0x01, 0x10, 0x00, 0xf0, 0xf1, 0x00, 0x30, 0x00, 0x70, 0x64, 0x00,
   0x20, 0x00, 0x30, 0x6e, 0xe0, 0x60, 0x00, 0x30, 0x6e, 0x40, 0x60, 0x00,
   0x30, 0x6e, 0x40, 0x60, 0x00, 0x70, 0x60, 0x40, 0x60, 0x00, 0xf0, 0x60,
   0x40, 0x60, 0x00, 0xf0, 0x71, 0x00, 0x20, 0x00, 0xe0, 0xff, 0x00, 0x30,
   0x00, 0x00, 0x0f, 0x01, 0x10, 0x00, 0x00, 0x00, 0x06, 0x0c, 0x00, 0x00,
   0x00, 0x0c, 0x07, 0x00, 0x00, 0x00, 0xf0, 0x01 };


Code/preliminary tests/v1/v1/p_40.h
#define p_40_width 40
#define p_40_height 40
static unsigned char p_40_bits[] = {
   0x00, 0x10, 0x00, 0x00, 0x00, 0x00, 0x10, 0x00, 0x00, 0x00, 0x00, 0x10,
   0x00, 0x07, 0x00, 0x10, 0x00, 0x10, 0x07, 0x00, 0x20, 0xfe, 0xc1, 0x02,
   0x00, 0x80, 0x83, 0x63, 0x03, 0x00, 0xc0, 0x01, 0x36, 0x01, 0x00, 0xc0,
   0x00, 0x94, 0x01, 0x00, 0x60, 0x00, 0xcc, 0x00, 0x00, 0x60, 0x00, 0xcc,
   0x00, 0x00, 0x20, 0x00, 0x6c, 0x00, 0x00, 0x2e, 0x7e, 0xe6, 0x07, 0x00,
   0xe0, 0xff, 0xf1, 0x1f, 0x00, 0xf0, 0xe7, 0x31, 0x3e, 0x00, 0xf0, 0xc0,
   0x10, 0x38, 0x00, 0x78, 0x60, 0x08, 0x70, 0x00, 0x38, 0x30, 0x08, 0xf0,
   0x00, 0x18, 0x30, 0x0c, 0xf0, 0x01, 0x1c, 0x18, 0x0c, 0xf0, 0x07, 0x0c,
   0x08, 0x06, 0xf0, 0x0f, 0x0c, 0x06, 0x03, 0x00, 0x0e, 0x0c, 0x07, 0x03,
   0x00, 0x0e, 0x8c, 0x83, 0x01, 0xf0, 0x1f, 0x8c, 0x81, 0x01, 0xfc, 0x1f,
   0xdc, 0xc0, 0x00, 0x0e, 0x1c, 0x7c, 0xc0, 0x00, 0x07, 0x10, 0x78, 0x60,
   0x80, 0x01, 0x30, 0x38, 0x60, 0xc0, 0x01, 0x60, 0x08, 0x30, 0x40, 0x00,
   0x40, 0x0c, 0xf0, 0x7f, 0x70, 0xc0, 0x0c, 0xf0, 0x7f, 0x90, 0xc0, 0x06,
   0xf8, 0x7f, 0x70, 0xc0, 0x02, 0xf8, 0x7f, 0x10, 0xc0, 0x33, 0x0c, 0x60,
   0x10, 0xc0, 0x33, 0x04, 0x40, 0x00, 0xc0, 0x33, 0x06, 0xc0, 0x00, 0x40,
   0x06, 0x03, 0x80, 0x01, 0x30, 0xcc, 0x01, 0x00, 0x03, 0x10, 0xf8, 0x00,
   0x00, 0x0e, 0x0e, 0x00, 0x00, 0x00, 0xf8, 0x03 };


Code/preliminary tests/v1_with_interrupt/rh_40.h
#define rh_40_width 40
#define rh_40_height 40
static unsigned char rh_40_bits[] = {
   0x00, 0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0xc0, 0x01, 0x00, 0x00, 0x00,
   0xe0, 0x01, 0x00, 0x00, 0xc0, 0x20, 0x02, 0x00, 0x00, 0xf0, 0x13, 0x04,
   0x00, 0x00, 0xf0, 0x13, 0x0c, 0x00, 0x00, 0xf0, 0x0f, 0x08, 0x00, 0x00,
   0x38, 0x0f, 0x08, 0x00, 0x00, 0x18, 0x06, 0x10, 0x00, 0x00, 0x1c, 0x06,
   0x10, 0x00, 0x00, 0x0e, 0x02, 0x30, 0x00, 0x00, 0x0e, 0x0b, 0x20, 0x00,
   0x80, 0x07, 0x0b, 0x20, 0x00, 0x80, 0x03, 0x12, 0x24, 0x00, 0xc0, 0x03,
   0x76, 0x33, 0x00, 0xe0, 0x01, 0xc6, 0x11, 0x00, 0xe0, 0x01, 0x0c, 0x10,
   0x00, 0xf0, 0x00, 0x08, 0x08, 0x00, 0x70, 0x00, 0xf0, 0x0f, 0x00, 0x70,
   0x00, 0xe0, 0x07, 0x00, 0x38, 0x00, 0x00, 0x0f, 0x00, 0x18, 0x00, 0x00,
   0x0e, 0x00, 0x18, 0x03, 0x00, 0x7e, 0x00, 0x18, 0x03, 0x00, 0xcf, 0x03,
   0x1c, 0x03, 0x80, 0x03, 0x06, 0x1c, 0x03, 0xc0, 0x00, 0x0c, 0x18, 0x03,
   0x40, 0x00, 0x08, 0x18, 0x03, 0x60, 0x00, 0x10, 0x18, 0x06, 0x30, 0x4e,
   0x12, 0x38, 0x0c, 0x30, 0x52, 0x12, 0x30, 0x18, 0x36, 0xce, 0x13, 0x70,
   0xf0, 0x37, 0x4a, 0x12, 0xf0, 0xc0, 0x30, 0x52, 0x12, 0xe0, 0x01, 0x30,
   0x00, 0x10, 0xc0, 0x07, 0x70, 0x00, 0x08, 0x80, 0x1f, 0xfe, 0x00, 0x08,
   0x00, 0xfc, 0x0f, 0x03, 0x06, 0x00, 0xf0, 0x03, 0xee, 0x03, 0x00, 0x00,
   0x00, 0x78, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };


Code/preliminary tests/v1_with_interrupt/voc_40.h
#define voc_40_width 40
#define voc_40_height 40
static unsigned char voc_40_bits[] = {
   0x70, 0x00, 0x00, 0x00, 0x00, 0xfc, 0x01, 0x00, 0x00, 0x00, 0x06, 0x03,
   0x00, 0x30, 0x00, 0x03, 0x02, 0x00, 0xfc, 0x00, 0x01, 0x06, 0x00, 0x86,
   0x01, 0x01, 0x04, 0x00, 0x02, 0x01, 0x01, 0x04, 0x00, 0x03, 0x02, 0x03,
   0x0e, 0x00, 0x0f, 0x01, 0x02, 0x0e, 0x80, 0x19, 0x01, 0x86, 0x11, 0xc0,
   0x90, 0x01, 0xfc, 0xf0, 0x67, 0xf8, 0x00, 0xc0, 0xe0, 0x3f, 0x0c, 0x00,
   0x80, 0xc0, 0x3f, 0x06, 0x00, 0x00, 0x81, 0x7d, 0x03, 0x00, 0x00, 0x83,
   0xf0, 0x01, 0x00, 0x00, 0xc6, 0xe0, 0x01, 0x00, 0x00, 0x7f, 0xe0, 0x01,
   0x00, 0x00, 0x3f, 0xc0, 0x01, 0x00, 0x00, 0x07, 0xc0, 0x01, 0x00, 0x00,
   0x07, 0xc0, 0x01, 0x00, 0x00, 0x07, 0xf0, 0x01, 0x00, 0x00, 0x07, 0xf0,
   0x01, 0x00, 0x00, 0x0f, 0x08, 0xe3, 0x03, 0x00, 0x1f, 0x08, 0x7e, 0x0e,
   0x80, 0x7f, 0x08, 0x0e, 0x18, 0x80, 0xfc, 0x1f, 0x06, 0x30, 0x40, 0xf8,
   0x3f, 0x03, 0x60, 0x78, 0xcc, 0xef, 0x01, 0xc0, 0x7c, 0x04, 0xc0, 0x00,
   0x80, 0x62, 0x02, 0x80, 0xc6, 0xb8, 0x43, 0x03, 0x40, 0x61, 0x85, 0x83,
   0x01, 0x40, 0xed, 0x99, 0x03, 0x01, 0x40, 0x29, 0xa1, 0x82, 0x01, 0x80,
   0x26, 0x9d, 0x86, 0x00, 0x80, 0x01, 0xc0, 0x7c, 0x00, 0x00, 0x03, 0x60,
   0x00, 0x00, 0x00, 0x02, 0x20, 0x00, 0x00, 0x00, 0x0c, 0x18, 0x00, 0x00,
   0x00, 0x38, 0x0e, 0x00, 0x00, 0x00, 0xe0, 0x03 };


Code/preliminary tests/display/bme680_200_mono.h
#define bme680_200_mono_width 200
#define bme680_200_mono_height 58
static unsigned char bme680_200_mono_bits[] = {
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0xe0, 0x0f, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0xc0, 0x01, 0x00, 0x00, 0x00, 0xe0, 0x1f, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x80, 0x03, 0x00, 0x00, 0xe0, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x80, 0x01, 0x00, 0xc6, 0x01, 0x00, 0x00, 0x00, 0xf0, 0x1e, 0x00, 0x00,
   0x00, 0x00, 0x00, 0xc0, 0x03, 0x00, 0x00, 0xf8, 0x07, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x83, 0x0f, 0xf3, 0x01, 0x00, 0x00, 0x00, 0x70, 0x1c, 0x00,
   0x00, 0x00, 0x00, 0xc0, 0x40, 0x04, 0x00, 0x00, 0x0c, 0x0c, 0x00, 0x80,
   0x01, 0x00, 0x00, 0xe2, 0x3f, 0xb0, 0x00, 0x00, 0x00, 0x04, 0x70, 0x1c,
   0x00, 0x00, 0x00, 0x00, 0xe0, 0x61, 0x0c, 0x00, 0x00, 0x06, 0x08, 0x00,
   0xe0, 0x07, 0x00, 0x00, 0x38, 0x70, 0xd8, 0x00, 0x00, 0x00, 0x04, 0x70,
   0x1c, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x33, 0x08, 0x00, 0x00, 0x02, 0x18,
   0x00, 0x30, 0x0c, 0x00, 0x00, 0x1c, 0xc0, 0x4c, 0x00, 0x00, 0x00, 0x04,
   0x70, 0x1c, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x13, 0x18, 0x00, 0x00, 0x02,
   0x10, 0x00, 0x10, 0x18, 0x00, 0x00, 0x0c, 0x80, 0x66, 0x00, 0x00, 0x80,
   0x3f, 0x70, 0x1e, 0x00, 0x00, 0x00, 0x00, 0xf8, 0x1f, 0x30, 0x00, 0x00,
   0x02, 0x10, 0x00, 0x18, 0x10, 0x00, 0x00, 0x06, 0x80, 0x33, 0x00, 0x00,
   0x80, 0x3f, 0x70, 0x1e, 0x00, 0x00, 0x00, 0x00, 0x3c, 0x0f, 0x20, 0x00,
   0x00, 0x02, 0x10, 0x00, 0x38, 0x10, 0x00, 0x00, 0x06, 0x80, 0x33, 0x00,
   0x00, 0x00, 0x04, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x00, 0x1c, 0x06, 0x60,
   0x00, 0x00, 0x06, 0x38, 0x00, 0x78, 0x18, 0x00, 0x00, 0x02, 0x80, 0x19,
   0x00, 0x00, 0x00, 0x04, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x00, 0x1e, 0x06,
   0x40, 0x00, 0x00, 0x04, 0x38, 0x00, 0xcc, 0x18, 0x00, 0x70, 0xe2, 0xcf,
   0xf8, 0x01, 0x00, 0x00, 0x04, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x00, 0x0f,
   0x02, 0xc0, 0x00, 0x00, 0x0c, 0x46, 0x00, 0x86, 0x0c, 0x00, 0x00, 0xfe,
   0x7f, 0xfc, 0x07, 0x00, 0x00, 0x00, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x00,
   0x07, 0x03, 0x80, 0x00, 0x00, 0xf8, 0xc3, 0x1f, 0xc3, 0x07, 0x00, 0x00,
   0xfe, 0x3f, 0xfc, 0x0f, 0x00, 0x00, 0x00, 0x70, 0x1c, 0x00, 0x00, 0x00,
   0x80, 0x07, 0x33, 0x80, 0x00, 0x00, 0x80, 0x83, 0xff, 0x61, 0x00, 0x00,
   0x00, 0x7f, 0x3c, 0x8e, 0x1f, 0x00, 0x00, 0x00, 0xf0, 0x1c, 0x00, 0x00,
   0x00, 0xc0, 0x03, 0x33, 0x80, 0x00, 0x00, 0x00, 0x02, 0xff, 0x31, 0x00,
   0x00, 0x80, 0x0f, 0x18, 0x06, 0x1e, 0x00, 0xf0, 0xff, 0xff, 0xff, 0xff,
   0x1f, 0x00, 0xc0, 0x01, 0x22, 0x80, 0x00, 0x00, 0x00, 0x04, 0xf6, 0x1b,
   0x00, 0x00, 0xc0, 0x07, 0x0c, 0x03, 0x3c, 0x00, 0x00, 0x00, 0xf0, 0x1f,
   0x00, 0x00, 0x00, 0xe0, 0x01, 0xe6, 0xc2, 0x00, 0x00, 0x00, 0x0c, 0x82,
   0x0f, 0x00, 0x00, 0xc0, 0x03, 0x06, 0x03, 0x78, 0x00, 0x00, 0x00, 0x70,
   0x1c, 0x00, 0x00, 0x00, 0xf0, 0x00, 0x86, 0x43, 0x00, 0x00, 0x00, 0x18,
   0x03, 0x0f, 0x00, 0x00, 0xc0, 0x01, 0x87, 0x01, 0xf8, 0x00, 0x00, 0x00,
   0x70, 0x1c, 0x00, 0x00, 0x00, 0xf0, 0x00, 0x0c, 0x60, 0x00, 0x00, 0x00,
   0xfc, 0x01, 0x0f, 0x00, 0x00, 0xe0, 0x81, 0x83, 0x01, 0xf8, 0x03, 0x00,
   0x00, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x78, 0x00, 0x18, 0x30, 0x00, 0x00,
   0x00, 0xfc, 0x00, 0x0e, 0x00, 0x00, 0xe0, 0x80, 0xc1, 0x00, 0xf8, 0x07,
   0x00, 0x00, 0x70, 0x1e, 0x00, 0x00, 0x00, 0x38, 0x00, 0xf0, 0x1f, 0x00,
   0x00, 0x00, 0x1c, 0x00, 0x0e, 0x00, 0x00, 0xe0, 0xc0, 0xc0, 0x00, 0x80,
   0x07, 0x80, 0x3f, 0x70, 0x1f, 0x00, 0x00, 0x00, 0x3c, 0x00, 0xc0, 0x0f,
   0x00, 0x00, 0x00, 0x1c, 0x00, 0x0e, 0x00, 0x00, 0xe0, 0x60, 0x60, 0x00,
   0x00, 0x0f, 0x00, 0x00, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x1c, 0x00, 0x00,
   0x0e, 0x00, 0x00, 0x00, 0x1c, 0x00, 0x0e, 0x00, 0x00, 0xe0, 0x70, 0x60,
   0x00, 0x00, 0x0f, 0x00, 0x00, 0x70, 0x1c, 0x00, 0x00, 0x00, 0x1e, 0x00,
   0x00, 0x1e, 0x00, 0x00, 0x00, 0x1c, 0x80, 0x0f, 0x00, 0x00, 0xe0, 0x38,
   0x30, 0x00, 0xf8, 0x1f, 0x00, 0x00, 0x70, 0x1c, 0xe0, 0x00, 0x00, 0x0e,
   0x00, 0x00, 0x1c, 0x00, 0x00, 0x00, 0x1c, 0xc0, 0x0f, 0x00, 0x00, 0xe0,
   0x18, 0x30, 0x00, 0xff, 0x1f, 0x00, 0x00, 0x70, 0x1c, 0xfc, 0x07, 0x00,
   0x8e, 0x01, 0x00, 0xfc, 0x01, 0x00, 0x00, 0x3c, 0x60, 0x18, 0x3f, 0x00,
   0xe0, 0x0d, 0x18, 0x80, 0x03, 0x1e, 0x00, 0x00, 0x70, 0x1c, 0x07, 0x0e,
   0x00, 0x8e, 0x01, 0x00, 0x3e, 0x0f, 0x00, 0x00, 0x7c, 0x60, 0xf0, 0xf3,
   0x00, 0xe0, 0x07, 0x18, 0xc0, 0x01, 0x18, 0x00, 0x00, 0x70, 0x9c, 0x01,
   0x18, 0x00, 0x8f, 0x01, 0x00, 0x07, 0x18, 0x00, 0x00, 0xfe, 0x61, 0x70,
   0x80, 0x01, 0xc0, 0x07, 0x0c, 0x60, 0x00, 0x30, 0x00, 0x00, 0x78, 0xfc,
   0x00, 0x30, 0x00, 0x8f, 0x01, 0x80, 0x01, 0x30, 0x00, 0x00, 0xf3, 0xff,
   0x30, 0x00, 0x03, 0xc0, 0x03, 0x0c, 0x70, 0x00, 0x60, 0x00, 0x00, 0x7c,
   0x78, 0x00, 0x60, 0x00, 0x8e, 0x01, 0x80, 0x00, 0x60, 0x00, 0x00, 0xe1,
   0xff, 0x19, 0x00, 0x06, 0x80, 0x01, 0x06, 0x30, 0x00, 0x60, 0x00, 0x00,
   0x1c, 0x70, 0x00, 0x40, 0x00, 0x8e, 0x01, 0xc0, 0x00, 0x40, 0x00, 0xf0,
   0x30, 0x7f, 0x0f, 0x00, 0x0c, 0xc0, 0x00, 0x06, 0x10, 0x00, 0x40, 0x00,
   0x00, 0x1e, 0x31, 0x00, 0x40, 0x00, 0x8e, 0x03, 0x40, 0x00, 0xc0, 0x00,
   0xd8, 0x10, 0x00, 0x06, 0x00, 0x08, 0x60, 0x00, 0xff, 0x1f, 0x38, 0xc0,
   0x00, 0x00, 0x8e, 0x33, 0xe0, 0xc0, 0x00, 0x0e, 0x03, 0x60, 0x9c, 0xc4,
   0x00, 0xc4, 0x08, 0x00, 0x64, 0x8c, 0x0b, 0x60, 0x00, 0xff, 0x1f, 0x48,
   0xc0, 0x00, 0x00, 0x8e, 0x37, 0x40, 0xc0, 0x00, 0x1e, 0x06, 0x60, 0xa4,
   0x84, 0x00, 0x86, 0x0d, 0x00, 0x14, 0x52, 0x08, 0x30, 0x80, 0xff, 0x1f,
   0x38, 0xc0, 0x00, 0x00, 0x8e, 0x33, 0x40, 0xc0, 0x00, 0x1c, 0x1c, 0x66,
   0x9c, 0x87, 0x00, 0x06, 0x07, 0x00, 0xd4, 0x9e, 0x09, 0x10, 0x80, 0xff,
   0x1f, 0x08, 0xc0, 0x00, 0x00, 0x1e, 0x30, 0x40, 0xc0, 0x00, 0x3c, 0xf8,
   0x67, 0x94, 0x84, 0x00, 0x06, 0x04, 0x00, 0x94, 0x12, 0x0a, 0x98, 0xc3,
   0x00, 0x18, 0x08, 0xc0, 0x00, 0x00, 0x3c, 0x30, 0x40, 0xc0, 0x00, 0x78,
   0xc0, 0x60, 0xa4, 0xc4, 0x00, 0x06, 0x04, 0x00, 0x64, 0xd2, 0x09, 0x98,
   0x43, 0x00, 0x10, 0x00, 0xc0, 0x00, 0x00, 0x7c, 0x3c, 0x00, 0x40, 0x00,
   0xf0, 0x00, 0x60, 0x00, 0xc0, 0x00, 0x04, 0x06, 0x00, 0x04, 0x00, 0x08,
   0x98, 0x63, 0x00, 0x30, 0x00, 0x40, 0x00, 0x00, 0xf8, 0x7f, 0x00, 0x60,
   0x00, 0xe0, 0x03, 0xf0, 0x00, 0x40, 0x00, 0x0c, 0x03, 0x00, 0x0c, 0x00,
   0x0c, 0x18, 0x30, 0x00, 0x20, 0x00, 0x60, 0x00, 0x00, 0xf0, 0xdf, 0x00,
   0x20, 0x00, 0xc0, 0x1f, 0xfe, 0x00, 0x60, 0x00, 0xf8, 0x01, 0x00, 0x18,
   0x00, 0x06, 0x30, 0x30, 0x00, 0x60, 0x00, 0x30, 0x00, 0x00, 0xc0, 0x87,
   0x00, 0x30, 0x00, 0x80, 0xff, 0xff, 0x01, 0x30, 0x00, 0x00, 0x00, 0x00,
   0x10, 0x00, 0x02, 0x60, 0x1c, 0x00, 0xc0, 0x00, 0x18, 0x00, 0x00, 0x00,
   0x00, 0x03, 0x1c, 0x00, 0x00, 0xfe, 0x1f, 0x06, 0x18, 0x00, 0x00, 0x00,
   0x00, 0x60, 0x80, 0x01, 0xc0, 0x0f, 0x00, 0x80, 0x01, 0x0c, 0x00, 0x00,
   0x00, 0x00, 0x0e, 0x07, 0x00, 0x00, 0xf0, 0x03, 0xbc, 0x0f, 0x00, 0x00,
   0x00, 0x00, 0xc0, 0xe1, 0x00, 0x00, 0x00, 0x00, 0x00, 0x8f, 0x07, 0x00,
   0x00, 0x00, 0x00, 0xf8, 0x01, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x01, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x3f, 0x00, 0x00, 0x00, 0x00, 0x00, 0xfc, 0x01,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
